gb-cpu/src/microcode/controller.rs

```diff
@@ -173,10 +169,18 @@ impl MicrocodeController {
         let current_dir_path = env::current_dir()?.into_os_string();
 
         if let Some(path) = current_dir_path.to_str() {
-            self.logs.push_str(opcode_logs);
+            let logs = match fs::read(format!("{}/debug/registers_logs/{}", path, "ours.txt")) {
```

Comment:
open the file with the [Append](https://doc.rust-lang.org/std/fs/struct.OpenOptions.html#method.append) option, that way  you don't have to read the file to append the line

Reply:
[fs::write](https://doc.rust-lang.org/std/fs/fn.write.html#) open and close the file each time you call it.
If you don't want to reopen the file at each iteration, you could store the result of `OpenOptions` in the struct